### PR TITLE
Created a git hook to warn the user if they messed up.

### PR DESCRIPTION
**Pull request by easyaspi314**
This will prevent the user from making files executable or not running
./de_after.sh.

Currently, the way we install the hook is in the Makefile in a rather dirty
way. It bugs me that pre-commit hooks can't be specified beforehand.

Basically, I put an empty file .GERMAN in `data-de/` and `graphics-de/`. These will be copied to `data` and `graphics` when you run `./de_before.sh`. If the user doesn't run `./de_after.sh`, `data/.GERMAN` and `graphics/.GERMAN` will exist and the pre-commit hook will fail.

Additionally, I check any added files for the executable permission aside from `.hooks/pre-commit`, and any shell scripts.

### CODE REVIEW

File: Makefile~ [hunk 1]
```diff
@@ -0,0 +1,174 @@
+
```
Comment (yenatch) [inline Makefile~]: this file should probably not be here

**Comment by easyaspi314** [comment]
Oh wow

**Comment by easyaspi314** [comment]
Remember when I said "wildcards are evil?"

**Comment by yenatch** [comment]
It would be much better to fix the german build so it can be run along the others rather than bandaiding it.

A hook for executable files is maybe something we could ask contributors to use, but it would just punish anyone using the repo on their own.

**Comment by easyaspi314** [comment]
Hmm. I wonder if it could be tested in Travis?

**Comment by yenatch** [comment]
Yes. That would be good.

**Comment by easyaspi314** [comment]
Then screw this.

**Pull request closed without merging**